basic_test.go

```diff
@@ -220,7 +220,7 @@ func TestIntegration(t *testing.T) {
 	}
 
 	records := make([]*record, 400)
-	tree := NewTree(nil, 0)
+	tree := NewMutableTree(db.NewMemDB(), 0)
```

Comment:
Why did this change?

Reply:
Posting what I said in the call here for future reference:

Anywhere in the tests where we were using `Tree` mutably I had to change it to `MutableTree`, and if you pass `nil` as the db to `MutableTree` it errors when constructing the nodedb. Using `MutableTree` with a MemDB as nodedb is equivalent to the old code using `Tree`.